tool/tsh/tsh.go

```diff
@@ -2965,6 +3131,110 @@ func onApps(cf *CLIConf) error {
 	return trace.Wrap(showApps(apps, profile.Apps, cf.Format, cf.Verbose))
 }
 
+type appListing struct {
+	Proxy   string            `json:"proxy"`
+	Cluster string            `json:"cluster"`
+	App     types.Application `json:"app"`
+}
+
+type appListings []appListing
+
+func (l appListings) Len() int {
+	return len(l)
+}
+
+func (l appListings) Less(i, j int) bool {
+	if l[i].Proxy != l[j].Proxy {
+		return l[i].Proxy < l[j].Proxy
+	}
+	if l[i].Cluster != l[j].Cluster {
+		return l[i].Cluster < l[j].Cluster
+	}
+	return l[i].App.GetName() < l[j].App.GetName()
+}
+
+func (l appListings) Swap(i, j int) {
+	l[i], l[j] = l[j], l[i]
+}
+
+func listAppsAllClusters(cf *CLIConf) error {
+	var listings appListings
+	err := forEachProfile(cf, func(tc *client.TeleportClient, profile *client.ProfileStatus) error {
+		result, err := tc.ListAppsAllClusters(cf.Context, nil /* custom filter */)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		for clusterName, apps := range result {
+			for _, app := range apps {
+				listings = append(listings, appListing{
+					Proxy:   profile.ProxyURL.Host,
+					Cluster: clusterName,
+					App:     app,
+				})
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	sort.Sort(listings)
+
+	profile, _, err := client.Status(cf.HomePath, "")
```

Comment:
As with database, I think this could be `client.StatusCurrent()`?

```suggestion
	profile, _, err := client.StatusCurrent(cf.HomePath, cf.Proxy, cf.IdentityFileIn)
```